### PR TITLE
[libunwind][AIX] Fix up TOC register if unw_getcontext is called from a different module

### PR DESCRIPTION
**Pull request by xingxue-ibm**
`unw_getcontext` saves the caller's registers in the context. However, if the caller of `unw_getcontext` is in a different module, the glue code of `unw_getcontext` sets the TOC register (r2) with the new TOC base and saves the original TOC register value in the stack frame. This causes the incorrect TOC value is used when the caller steps up frames, which fails libunwind LIT test case `unw_resume.pass.cpp`. This PR fixes the problem by using the original TOC register value saved in the stack if the caller is in a different module and enables `unw_resume.pass.cpp` on AIX.

### CODE REVIEW

File: libunwind/src/UnwindRegistersSave.S [hunk 1]
```diff
@@ -305,9 +305,22 @@ DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
   mflr  0
   std   0, PPC64_OFFS_SRR0(3) // store lr as ssr0
   PPC64_STR(1)
+  PPC64_STR(4)      // Save r4 first since it will be used for fixing r2.
+#if defined(_AIX)
+  // The TOC register (r2) was changed by the glue code if unw_getcontext
+  // is called from a different module. Save the original TOC register
+  // in the context if this is the case.
+  mflr  4
```
Comment (stephenpeckham) [inline libunwind/src/UnwindRegistersSave.S:313]: Not a bit deal, but LR is already in r0.  Do you need mflr twice? You could use mr r4,r0.

File: libunwind/src/UnwindRegistersSave.S [hunk 2]
```diff
@@ -305,9 +305,22 @@ DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
...
+  lwz   4, 0(4)     // Get the first instruction at the return address.
+  lis   0, 0xe841   // Is it reloading the TOC register "ld 2,40(1)"?
```
Comment (stephenpeckham) [inline libunwind/src/UnwindRegistersSave.S:315]: Why is the 32- and 64-bit code different?  Can't you use the same comparison technique?

Comment (stephenpeckham) [reply libunwind/src/UnwindRegistersSave.S:315]: The legacy compiler uses xoris/cmplwi for both 32-bit and 64-bit.

File: libunwind/src/UnwindRegistersSave.S (hunk 1 above)
Comment (stephenpeckham) [reply libunwind/src/UnwindRegistersSave.S:313]: Either way is fine with me.

File: libunwind/src/UnwindRegistersSave.S (hunk 2 above)
Comment (xingxue-ibm) [reply libunwind/src/UnwindRegistersSave.S]: Changed to use the same comparison technique for 32- and 64-bit as suggested, thanks!

**Review by stephenpeckham — COMMENTED** [review]
No further comments

**Review by stephenpeckham — APPROVED** [review]

**Pull request merged**